bin/sp-data-pull

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'json'
+
+data_file = 'your-json-file-here.json'
+
+imported_data = File.read(data_file)
+
+parsed_data = JSON.parse(imported_data)
+
+user_uuids = parsed_data.map { |user| user[:"properties.user_id"] }
+
+# collects user information per scenario and helps collect median information
+no_sp_users = Array.new
+single_sp_users = Array.new
+mulitple_sp_users = Array.new
+
+# collect information on service provider per scenario
+single_sp_list = Set.new
+multiple_sp_list = Set.new
+
+# count of service probiders
+single_sp_tally  = 0
+multiple_sp_tally = 0
+
+User.includes(:service_providers).where(uuid: user_uuids).in_batches(of: 1_000) do |batch|
+  # array of users
+  no_sp_users = batch.select { |user| user.service_providers.blank? }
+  single_sp_users = batch.select { |user| user.service_providers.count == 1 }
+  multiple_sp_users = batch.select { |user| user.service_providers.count > 1 }
+
+  # list of service prooviders per scenario
+  single_sp_users.each { |user| single_sp_list << user.service_providers.map { |u| u.friendly_name } }
+  multiple_sp_users.each { |user| multiple_sp_list << user.service_providers.map { |u| u.friendly_name } }
+
+  # sp tally
+  single_sp_tally = single_sp_list.to_a.flatten.tally
+  multiple_sp_tally = multiple_sp_list.to_a.flatten.tally
+
+  sleep(0.2)
+end
```

Comment:
there might be a way to get a lot of this info by doing 2 aggregate queries?

```
SELECT
  COUNT (*) AS num_users
 , sps_per_user.num_sps
FROM
(
  SELECT COUNT(*) AS num_sps, user_id
  FROM identities
  GROUP BY user_id
) sps_per_user
```
^^ gives the number of users with # of SPs, could be converted to do the > 1 / == 1 bit

And then another query to get the number of users with no SP

```
SELECT COUNT(*)
FROM users
WHERE users.id NOT IN (
  SELECT user_id FROM service_providers
)
```

Reply:
This has me thinking--instead of this being a script, would this work be better suited to something similar as a report?

Reply:
possibly! I don't know the audience for the report but that does make sense, especially if it's recurring

Reply:
I think a report would be valuable if we'd be looking to get this on a regular basis, but as a one-off, I think we can avoid the overhead.